### PR TITLE
LKE-8090 fix(queryParsing): use URLSearchParams to get url query params

### PR DESCRIPTION
**Pull request by aminerhanemi**

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -111,57 +111,56 @@ async function runQueryByID(query) {
...
-            if (parameter.type === 'global') {
-                result.global[decodeURIComponent(parameter.key)] = parameter.value;
-            } else if (parameter.type === 'templateField') {
-                result.templateFields[decodeURIComponent(parameter.key)] = parameter.value;
-            }
-        }
-    });
-    queryParams = result;
+  const result = {global: {}, templateFields: {}};
+  const parameters = new URLSearchParams(location.search);
```
Comment (hsenbassam) [inline public/js/main.js:115]: Do we have any concern related to the browser compatibility of URLSearchParams ? https://caniuse.com/urlsearchparams

I don't see incompatibility but just making sure that it is not working with IE and some old versions and if it is okay

File: public/js/main.js [hunk 2]
```diff
@@ -111,57 +111,56 @@ async function runQueryByID(query) {
...
+      'param_string_'
+    ]
+  };
+  const decodedValue = decodeURIComponent(value);
+  if (allowedParams.oneOf.includes(key)) {
+    return {key: key, value: decodedValue, type: 'global'};
+  } else {
+    const prefix = allowedParams.startWith.find((prefix) => key.startsWith(prefix));
+    switch (prefix) {
+      case  'param_number_':
```
[the remark this answers is not in the document]
Comment (EricMallard) [reply public/js/main.js:154]: this one is a bit more blocking, I know it's legacy but it seems like a no brainer improvement (if I did not miss something that prevents us to implement it)

File: public/js/main.js (hunk 1 above)
Comment (aminerhanemi) [reply public/js/main.js:115]: no we don't support EI and for others
last 3 Firefox major versions
last 3 Chrome major versions
last 3 ChromeAndroid major versions
last 1 Edge major versions
last 1 Safari major versions
last 1 iOS major versions
last 1 Opera major versions

so it should be fine. (we are also using it in the frontend)

File: public/js/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (aminerhanemi) [reply public/js/main.js:154]: there is also converting decodeValue when it is a number using + .
If you use +'someString' it will be converted to NaN.

**Review by hsenbassam — APPROVED** [review]

**Review by EricMallard — APPROVED** [review]

**Pull request merged**